modules/oe_subscriptions_anonymous/oe_subscriptions_anonymous.tokens.inc

```diff
@@ -0,0 +1,78 @@
+<?php
+
+/**
+ * @file
+ * Token callbacks for the Anonymous subscriptions module.
+ */
+
+declare(strict_types=1);
+
+use Drupal\Core\Render\BubbleableMetadata;
+use Drupal\Core\Url;
+
+/**
+ * Implements hook_token_info().
+ */
+function oe_subscriptions_anonymous_token_info(): array {
+  $type = [
+    'name' => t('Anonymous subscriptions'),
+    'description' => t('Tokens related to anonymous subscriptions.'),
+  ];
+
+  $anonymous_subscriptions['subscriptions_page'] = [
+    'name' => t('Subscriptions page'),
+    'description' => t('Link to subscriptions page based on message owner, to be used in message entities.'),
+    'needs-data' => 'message',
+  ];
+
+  return [
+    'types' => ['oe_subscriptions_anonymous' => $type],
+    'tokens' => ['oe_subscriptions_anonymous' => $anonymous_subscriptions],
+  ];
+}
+
+/**
+ * Implements hook_tokens().
+ */
+function oe_subscriptions_anonymous_tokens(string $type, array $tokens, array $data, array $options, BubbleableMetadata $bubbleable_metadata) {
+  $replacements = [];
+
+  if (
+    $type == 'oe_subscriptions_anonymous' &&
+    isset($tokens['subscriptions_page']) &&
+    !empty($data['message'])
+  ) {
```

Comment:
I like this formatting better:
```suggestion
  if ($type === 'oe_subscriptions_anonymous'
    && isset($tokens['subscriptions_page'])
    && !empty($data['message'])
  ) {
```
Actually, Drupal core is inconsistent about this.
And in /vendor/ you also find different formats.
See
https://www.drupal.org/project/coding_standards/issues/3392321
https://www.drupal.org/project/coding_standards/issues/1539712
Look for "&&" to find relevant comments.

Ultimately you can decide this with the OEL team.